src/plugins/management/public/components/management_app_wrapper/management_app_wrapper.tsx

```diff
@@ -58,11 +81,16 @@ export class ManagementAppWrapper extends Component<ManagementSectionWrapperProp
 
   render() {
     return (
-      <div
-        // The following classes are a stop-gap for this element that wraps children of KibanaPageTemplate
-        className={classNames('euiPageContentBody', APP_WRAPPER_CLASS)}
-        ref={this.mountElementRef}
-      />
+      <KibanaErrorBoundaryProvider>
+        <KibanaErrorBoundary>
```

Comment:
> I'm thinking would be great to move it out of this component—higher into React tree, so that it catches more errors, for example, the errors that happen in the <ManagementAppWrapper> component directly. Was there a problem adding somewhere around the "routes" components?

Using this suggestion works. The Error Boundary can live in the route component. The error needs to be explicitly thrown from the app wrapper (not sure why) and I chose to do that with the `RecallError` (naming TBD) component.

> we could create an app aware error boundary component, which would know the meta information about the app, so it could render nicer errors:

I think this is a great suggestion. It is difficult to get alignment on the end user experience for each of the error scenarios, but this suggestion is a good conversation-driver.